src/test/java/uk/gov/hmcts/reform/sandl/snlevents/service/SessionServiceTests.java

```diff
@@ -0,0 +1,320 @@
+import java.time.LocalDate;
+import java.time.OffsetDateTime;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.AdditionalAnswers.returnsFirstArg;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(SpringRunner.class)
+@Import(JpaTestConfiguration.class)
+public class SessionServiceTests {
+    @Mock
+    EntityManager entityManager;
+
+    @Before
+    public void init() {
+        EntityManager em = mock(EntityManager.class);
+        sessionService.entityManager = em;
+
+        when(sessionRepository.save(any(Session.class))).then(returnsFirstArg());
+    }
+
+    private static final OffsetDateTime OFFSET_DATE_TIME = OffsetDateTime.MAX;
+    private static final long DURATION = 1L;
+    private static final String CASE_TYPE = "case-type";
+    private static final String UUID_STRING = "38400000-8cf0-11bd-b23e-10b96e4ef00d";
+    private static final String JUDGE_NAME = "judge-name";
+    public static final LocalDate START_DATE = LocalDate.MIN;
+    public static final LocalDate END_DATE = LocalDate.MAX;
+
+    @TestConfiguration
+    static class SessionServiceTestContextConfiguration {
+        @Bean
+        public SessionService sessionService() { return new SessionService(); }
+    }
+
+    @Autowired
+    private SessionService sessionService;
+
+    @MockBean
+    private SessionRepository sessionRepository;
+    @MockBean
+    private RoomRepository roomRepository;
+    @MockBean
+    private PersonRepository personRepository;
+    @MockBean
+    private HearingPartRepository hearingPartRepository;
+    @MockBean
+    private UserTransactionService userTransactionService;
+    @MockBean
+    private ObjectMapper objectMapper;
+    @MockBean
+    private FactsMapper factsMapper;
+    @MockBean
+    private RulesService rulesService;
+
+    @Test
+    public void getSessions_returnsSessionsFromRepository() {
+        List<Session> repositorySessions = createSessions();
+        when(sessionRepository.findAll()).thenReturn(repositorySessions);
+
+        List serviceSessions = sessionService.getSessions();
+
+        assertThat(serviceSessions).isEqualTo(repositorySessions);
+    }
+
+    @Test
+    public void getSessionById_returnsSessionFromRepository() {
+        Session repositorySession = createSession();
+        when(sessionRepository.findOne(any(UUID.class))).thenReturn(repositorySession);
+
+        Session serviceSession = sessionService.getSessionById(UUID.randomUUID());
+
+        assertThat(serviceSession).isEqualTo(repositorySession);
+    }
+
+    //@Test
+    //todo something wrong with mocking typed query
```

Comment:
https://stackoverflow.com/questions/7500312/mockito-match-any-class-argument  but from few tries, I can see it's not working.

Reply:
still I will like to see this query being built in repository rather than in service, so we better discuss moving it than fixing this test